### PR TITLE
Move around rake tasks

### PR DESCRIPTION
**Pull request by kddnewton**
Just some small organization.

* Split out memcheck stuff into `test.rake`
* Split out RDoc stuff into `rdoc.rake`
* Split out bear stuff into `bear.rake`
* Rename `tasks` to `rakelib` (the default for rake)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -2,11 +2,11 @@
 
 source "https://rubygems.org"
 
-ruby ">= 3.1.0"
+ruby ">= 3.0.0"
```
Comment (jemmaissroff) [inline Gemfile:5]: Why this change?

Comment (kddnewton) [reply Gemfile]: @jemmaissroff incorrectly committed on this branch. I reverted

**Review by jemmaissroff — APPROVED** [review]
